modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,192 @@
++
+////
+TODO: I skipped a step that did this:
+oc extract cm/default-ingress-cert -n openshift-config-managed --to=my-directory
+I assume we only did this because we're running keycloak from OCP, and we wouldn't do this if their provider was elsewhere, correct?
+////
+
+. Create a config map that contains the provider's certificate authority bundle by running the following command:
++
+[source,terminal]
+----
+$ oc create configmap keycloak-oidc-ca --from-file=ca-bundle.crt=my-directory/ca-bundle.crt \// <1>
+    -n openshift-config
+----
+<1> Specify the path to your provider's `ca-bundle.crt` file.
+
+. Edit the authentication configuration by running the following command:
++
+[source,terminal]
+----
+$ oc edit authentication.config/cluster
+----
+
+. Update the authentication configuration by setting the `type` field to `OIDC`, configuring the `oidcProviders` field for your Keycloak provider, and setting the `webhookTokenAuthenticator` field to `null`:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+# ...
+spec:
+  oidcProviders: <1>
+  - claimMappings:
+      groups:
+        claim: groups <2>
+        prefix: 'oidc-groups-test:'
+      username:
+        claim: email <3>
+        prefixPolicy: Prefix
+        prefix:
+          prefixString: 'oidc-user-test:'
+    issuer:
+      audiences: <4>
+      - console-test
+      - oc-cli-test
+      issuerCertificateAuthority:
+        name: keycloak-oidc-ca <5>
+      issuerURL: https://keycloak-keycloak.apps.example.com/realms/master <6>
+    name: 'keycloak-oidc-server' <7>
+    oidcClients:
+    - clientID: oc-cli-test <8>
+      componentName: cli
+      componentNamespace: openshift-console
+    - clientID: console-test <9>
+      clientSecret:
+        name: console-secret <10>
+      componentName: console
+      componentNamespace: openshift-console
+  type: OIDC <11>
+  webhookTokenAuthenticator: null <12>
+----
+<1> The OIDC provider configuration.
+<2> The name of the claim to construct group names for the cluster identity.
+<3> The name of the claim to construct usernames for the cluster identity.
+<4> The list of audience IDs that this authentication provider issues tokens for.
+<5> The name of the config map that contains the `ca-bundle.crt` key. If unset, system trust is used instead.
+<6> The URL for the token issuer.
+<7> The name for external OIDC provider.
+<8> The client ID that your provider uses for the {oc-first}.
+<9> The client ID that your provider uses for the {product-title} web console.
+<10> The name of the secret that stores the secret value for the console client.
+<11> A value of `OIDC` indicates to use an external OIDC identity provider.
+<12> Must be set to `null` when `type` is set to `OIDC`.
++
+//TODO what does end of #5 mean? https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-79807 - don't see the difference in config for issuerCertificateAuthority being unset
```

Comment:
@liouk @xingxingxia Can you clarify what the second sentence means for callout 5? About if you leave off the config map with the ca-bundle.crt, what system trust being used means?

Reply:
@bergerhoffer , system trust means /etc/pki/ , it is as same as:
```
$ oc explain oauth.spec.identityProviders.openID.ca | grep system
    If empty, the default system roots are used.
```
https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/authentication_and_authorization/configuring-identity-providers#configuring-oidc-identity-provider has a reference callout for the configmap of CA